### PR TITLE
Swap uvisor and uvisor lib

### PR DESCRIPTION
**Pull request by AlessandroA**
With this PR ARMmbed/uvisor-lib is not a submodule of ARMmbed/uvisor any more.
See ARMmbed/uvisor-lib#39 for the related changes in uvisor-lib.

What changed:
* Not just the uVisor binary, but the whole uVisor library is now pre-compiled. The exported uVisor is now a static library.
* All APIs + exported header files have been placed in the top-level `api` folder.

Note: Libraries are released in the `api/lib` folder. It has not been pushed yet, as it will be done when the PR is merged.

@meriac @Patater

### CODE REVIEW

File: api/inc/halt_exports.h [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016, ARM Limited, All Rights Reserved
+ * Copyright (c) 2015, ARM Limited, All Rights Reserved
```
Comment (meriac) [inline api/inc/halt_exports.h]: @AlessandroA 2015 looks wrong - should be 2016 - or rather "20whatever-2016"

Comment (AlessandroA) [inline api/inc/halt_exports.h]: Sorry, that ended up there by mistake. Rebased to have 2016.

**Comment by AlessandroA** [comment]
* Removed commit that moved `platform` inside `core/`

@meriac

**Comment by AlessandroA** [comment]
* Added a commit that removes the `vIRQ_SetVectorX` API. This commit is related to ARMmbed/uvisor-lib#41

**Comment by meriac** [comment]
@AlessandroA Please move https://github.com/ARMmbed/uvisor/pull/189/commits/b7fb61b12e821d0b392cd1237e5ff4b4337e6e0f out and submit independently. I want to discuss that separately. The rest LGTM.

**Comment by AlessandroA** [comment]
@meriac Removed commit about `g_nvic_prio_bits`.

**Pull request merged**